detectable/src/main/java/com/synopsys/integration/detectable/detectables/pnpm/lockfile/model/PnpmPackageInfov6.java

```diff
@@ -0,0 +1,55 @@
+package com.synopsys.integration.detectable.detectables.pnpm.lockfile.model;
+
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.commons.collections4.MapUtils;
+import org.jetbrains.annotations.Nullable;
+
+public class PnpmPackageInfov6 {
+    @Nullable
+    public Boolean dev;
+    @Nullable
+    public Boolean optional;
+    @Nullable
+    public Map<String, String> dependencies;
+    @Nullable
+    public Map<String, String> optionalDependencies;
+    @Nullable
+    public Map<String, String> devDependencies;
+    @Nullable
```

Comment:
The new optionalDependencies and devDependencies here will allow us to not miss as many transitive dependencies if the user does not explicitly exclude them. For now we are doing this only for v6 so we needed a new class.